src/Network/Minio/Data.hs

```diff
@@ -242,6 +242,34 @@ data ObjectInfo = ObjectInfo {
   , oiSize :: Int64
   } deriving (Show, Eq)
 
+data GetObjectOptions = GetObjectOptions {
+    -- | (0, 9) means first ten bytes of the source object
+    gooRange :: Maybe (Int64, Int64)
```

Comment:
We should change this to `Maybe Network.HTTP.Types.ByteRange`. I just checked that both S3 and Minio supports range headers like:

`Range: bytes=-100` -> first 101 bytes
`Range: bytes=90-100` -> 11 bytes starting from the 91st byte
`Range: bytes=100-` -> All remaining bytes starting from the 101st byte

The `http-types` package has just the appropriate type for this as indicated above.

S3 also documents that multiple ranges are not supported - quoting from https://docs.aws.amazon.com/AmazonS3/latest/API/RESTObjectGET.html:

> Amazon S3 doesn't support retrieving multiple ranges of data per GET request.

Reply:
Isn't it too low level? to keep it Network.HTTP.Types.ByteRange ?